rmw_dds_common/src/graph_cache.cpp

```diff
@@ -967,6 +967,13 @@ GraphCache::get_node_names(
         if (!node_namespaces->data[j]) {
           goto fail;
         }
+        if (security_contexts) {
```

Comment:
@ivanpauno use without prior initialization? Why not doing the same we do for `node_names` and `node_namespaces`? Or at least check it has been initialized and there's enough room for all elements.

Reply:
Oh, that was a good one. See https://github.com/ros2/rmw_dds_common/pull/4/commits/b7f8ce4e6986b96a49838a562510e3266d3a4222.

Reply:
bottom line: I have to add tests in `rcl`.